src/Indigo/Contracts/HomebaseLite/Impl/Admin.hs

```diff
@@ -0,0 +1,76 @@
+-- SPDX-FileCopyrightText: 2022 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+
+{-# OPTIONS_GHC -Wno-redundant-constraints -Wno-orphans #-}
+
+module Indigo.Contracts.HomebaseLite.Impl.Admin
+  ( setAdmin
+  , acceptAdmin
+  , addMaintainers
+  , removeMaintainers
+  , configure
+  ) where
+
+import Indigo hiding ((*))
+
+import Morley.Util.Interpolate (itu)
+
+setAdmin :: tp :~> KeyHash => IndigoEntrypoint tp
+setAdmin _ = do
+  description [itu|
+    Called by the current `admin` to transfer its role to someone else.
+    Takes in input the address of the new `admin` candidate.
+    Note: for security reasons the transfer isn't complete until the new admin candidate calls `accept_admin`.
+
+    - fails when the sender is not the current `admin`
+    - if there is already a candidate, another call to this entrypoint will replace it
+    - if there is already a candidate, calling this with the `admin` address will invalidate the current candidate
+    |]
+  pass
+
+
+acceptAdmin :: tp :~> () => IndigoEntrypoint tp
```

Comment:
Oh boy, I forgot how uncomfortable these signatures are to use and look at, I'll take a note to improve Indigo on it.

Reply:
FWIW, I've defined a synonym for this unpleasantness in #8, which makes it slightly less horrible.